src/server/api/v1/utils.py

```diff
@@ -0,0 +1,170 @@
+    Default handler called when client tries to access non-existed endpoint
+    :param req: request object
+    :param args:
+    :param kwargs:
+    :return: server reply with status code 'Not found' and message
+    """
+    req.serv_reply(404, {'message': 'Requested path is not found, please check the endpoint and try again'})
+
+
+def login(req, data):
+    """
+    Functionality to perform login actions.
+    Possible use cases:
+    1. Provided correct credentials and user exists in DB -> token is generated and user's db data updated
+    2. Provided incorrect credentials - 401 error returned
+    3. Provided only username or only password - 401 error returned
+    :param req: request object
+    :param data: dict with provided username and password
+    :return: user's token in case of success, 'Unauthorized' error in case if credentials validation failed
+    """
+    user = db_helper.UserDBHelper.user_exists(data)
+    if not (data.get('nickname') and data.get('password')):
+        req.serv_reply(401, {'message': 'Unauthorized. Username or password is invalid'})
+
+    elif not (user and check_password_hash(user.passwd, data.get('password'))):
+        req.serv_reply(401, {'message': 'Unauthorized. Username or password is invalid'})
+    else:
+        token = encode_auth_token(user.id)
+
+        # data to be updated in the DB {<field_name>: <value>}
+        data = {'token': token}
+        db_helper.UserDBHelper.update(user.id, data)
+        db_helper.db_connection.commit()
+        req.serv_reply(200, {'token': token.decode('utf-8')})
+
+
+def register(req, data):
+    """
+    New user registration procedure.
+    Possible use cases:
+    1. User provides credentials and user does not exist - new user registered
+    2. User provides the nickname which is already exists in the DB - 'Conflict' status returned
+    3. User does not provided whether username or password - 'Bad request' status returned
+    :param req: request object
+    :param data: dict with provided username and password
+    :return: 'Created' status code in case of successful registration, 'Bad request' or 'Conflict' in case of
+            credentials validation failed
+    """
+    if not (data.get('nickname') and data.get('password')):
+        req.serv_reply(400, {'message': 'Username and password are mandatory'})
+    elif not db_helper.UserDBHelper.user_exists(data):
+        user_obj = db_helper.UserDBHelper(db_helper.db_connection)
+        passwd = generate_password_hash(data.get('password'))
+        user_obj.create(data.get('nickname'), passwd)
+        req.serv_reply(201, {'message': 'User successfully created'})
+    else:
+        req.serv_reply(409, {'message': 'Username is already taken'})
+
+
+def users(req):
+    """
+    Get all users list
+    :param req: request object
+    :return: json with all registered users
+    """
+    users = db_helper.db_connection.query(UserModel).all()
+    response = {'users': []}
+    for user in users:
```

Comment:
How about exclude yourself from this list ?

Reply:
I do not understand your question. Could you explain more detailed what do you mean?

Reply:
response['users'] = map(lambda user: {'id': user.id, 'name': user.nickname}, users) 

because better to collect them all and then add to dict instead of 
calling value by key for each user